enterprise_access/apps/content_assignments/models.py

```diff
@@ -323,12 +323,16 @@ def annotate_dynamic_fields_onto_queryset(cls, queryset):
             QuerySet: LearnerContentAssignment queryset, same objects but with extra fields annotated.
         """
         # Annotate a derived field ``recent_action_time`` using pure ORM so that we can order_by() it later.
-        # ``recent_action_time`` is defined as the time of the most recent reminder, and falls back to assignment
-        # creation time if there are no reminders.
+        # ``recent_action_time`` is defined as the time of the most recent successful reminder, and falls
+        # back to assignment creation time if there are no successful reminders.
         new_queryset = queryset.annotate(
             recent_action_time=Coalesce(
                 # Time of most recent reminder.
-                Max('actions__completed_at', filter=Q(actions__action_type=AssignmentActions.REMINDED)),
+                Max('actions__completed_at', filter=(
+                    Q(actions__action_type=AssignmentActions.REMINDED) &
+                    Q(actions__error_reason__isnull=True) &
+                    Q(actions__completed_at__isnull=False)
```

Comment:
Before, this logic would consider reminder actions even when they have an `error_reason` and/or have not been marked as "completed" via the `completed_at` field.